File: 0058-length-of-last-word/0058-length-of-last-word.java
class Solution {
    public int lengthOfLastWord(String s) {
        boolean trimDone = false;
        int count = 0;
        for (int i = s.length()-1; i >= 0; i--){
            if (s.charAt(i)!= ' '){
                count++;
                trimDone = true;
            } else if (trimDone) return count;
        }
        return count;
    }
}

File: 0058-length-of-last-word/0058-length-of-last-word.ts
function lengthOfLastWord(s: string): number {
    let length = 0
    let idx = 0
    for (let i = s.length-1; i >= 0; i--){
        if (s[i] !== " "){
            idx = i
            break;
        }
    }
    for (let i = idx; i >= 0; i--){
        if (s[i] == " "){
            break;
        }
        length = length + 1
    }
    return length
};